chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -945,75 +988,37 @@ impl<'a, S: BlockchainStorageWrite, V: TransactionVerificationStrategy> Chainsta
         Ok(prev_block_index)
     }
 
+    // Returns true if a reorg has occurred.
     pub fn activate_best_chain(
         &mut self,
-        new_block_index: BlockIndex,
-        best_block_id: Id<GenBlock>,
-    ) -> Result<Option<BlockIndex>, BlockError> {
+        new_block_index: &BlockIndex,
+        best_block_id: &Id<GenBlock>,
+    ) -> Result<bool, BlockError> {
         // Chain trust is higher than the best block
         let current_best_block_index = self
-            .get_gen_block_index(&best_block_id)
+            .get_gen_block_index(best_block_id)
             .map_err(BlockError::BestBlockLoadError)
             .log_err()?
             .expect("Inconsistent DB");
 
         if new_block_index.chain_trust() > current_best_block_index.chain_trust() {
-            self.reorganize(&best_block_id, &new_block_index).log_err()?;
-            return Ok(Some(new_block_index));
+            self.reorganize(best_block_id, new_block_index).log_err()?;
+            return Ok(true);
```

Comment:
Why did you change this to return a boolean instead of the new tip's block index?

Reply:
It's moved. Got it. All good.